### PR TITLE
Add login and logout - backend and default frontend

### PR DESCRIPTION
**Pull request by ghiocgeorge**
Now if you want to have access for Dashboard, you must to login. When you are logged, the logout button will appear. In the future I will change the menu functionality, but for now I only put this logout button so we can check the full functionality.

### CODE REVIEW

File: src/main/java/ro/vladutit/Don/t/forget/v2/repository/UserRepository.java [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 import ro.vladutit.Don.t.forget.v2.model.User;
 
 @Repository
 public interface UserRepository extends CrudRepository<User, Long> {
+
+    @Query("SELECT u FROM User u WHERE u.email = ?1")
```
Comment (andreisamoila74) [inline src/main/java/ro/vladutit/Don/t/forget/v2/repository/UserRepository.java:11]: why do you have sql code here??

**Review by andreisamoila74 — CHANGES_REQUESTED** [review]

**Comment by andreisamoila74** [comment]
"When you are log in," -> logged*

File: src/main/java/ro/vladutit/Don/t/forget/v2/repository/UserRepository.java (hunk 1 above)
Comment (ghiocgeorge) [reply src/main/java/ro/vladutit/Don/t/forget/v2/repository/UserRepository.java:11]: I read these days about Queries on the Spring documentation and from what I understand, it's that it works very well without this line for a small number of queries, but I think that in the future, when the application grows, this line will help . I read this here [LINK](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.at-query). What do you think?

Comment (andreisamoila74) [reply src/main/java/ro/vladutit/Don/t/forget/v2/repository/UserRepository.java:11]: ok then, good catch👍🏻

Comment (ghiocgeorge) [reply src/main/java/ro/vladutit/Don/t/forget/v2/repository/UserRepository.java:11]: Thank you!

**Pull request merged**